postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py

```diff
@@ -0,0 +1,387 @@
+args = parser.parse_args()
+
+def LoadSingleComponentParallel(component, surfaceData=True):
+    chunks = np.int_(np.linspace(0, nElements, nprocs+1)) if surfaceData else np.int_(np.linspace(0, nElementsFault, nprocs+1))
+    chunks = np.array([chunks[:-1], chunks[1:]-chunks[:-1], np.full(nprocs, component)]).T
+    chunks = [tuple(i) for i in chunks]
+    
+    if surfaceData:
+        p = Pool(nprocs)
+        result = p.map(ReadSurfaceWithPool, chunks)
+        p.close()
+        p.join()
+        return np.concatenate(result, axis=1).T[::stepsize]
+    else:
+        p = Pool(nprocs)
+        result = p.map(ReadFaultWithPool, chunks)
+        p.close()
+        p.join()
+        return np.concatenate(result, axis=1).T[::stepsize]
+    
+def ReadSurfaceWithPool(chunk):
+    return surfacexdmf.ReadDataChunk(chunk[2], int(chunk[0]), int(chunk[1]))[timeIndices[0]:timeIndices[1]]
+
+def ReadFaultWithPool(chunk):
+    return faultxdmf.ReadDataChunk(chunk[2], int(chunk[0]), int(chunk[1]))[timeIndicesFault[0]:timeIndicesFault[1]]
+
+def ComputeTriangleMidpoints(geom, connect):
+    """Generates an array with coordinates of triangle midpoints (same dimension as connect)"""
+    xyz = np.zeros_like(connect, dtype=float)
+    xyz = (1./3.)*(geom[connect[:,0],:]+geom[connect[:,1],:]+geom[connect[:,2],:])   
+    return xyz
+
+def CalculateSlipCentroid(faultxdmf, events=0):
+    ASl = faultxdmf.ReadData("ASl", idt=timeIndicesFault[1]-1).T
+    if timeIndicesFault[0]!=0:
+        ASl -= faultxdmf.ReadData("ASl", idt=timeIndicesFault[0]).T
+    return np.average(faultxyz, axis=0, weights=ASl)
+
+def ComputeBackazimuth(xyz, centroid):
+    ba = np.arctan2(xyz[:,0]-centroid[0], xyz[:,1]-centroid[1]) % (2*np.pi)
+    ba = np.where(ba < np.pi, ba+np.pi, ba-np.pi)
+    return ba
+
+def RotateToRadialTransversal(u, v, xyz, centroid):
+    ba = ComputeBackazimuth(xyz, centroid)
+    ba = ba.reshape(-1,1)
+    r = -u * np.sin(ba) - v * np.cos(ba)
+    t = -u * np.cos(ba) + v * np.sin(ba)
+    return r, t
+
+def ComputeCornerFrequencyParallel(component, pphases, windowduration):
+    chunk_inds = np.linspace(0, component.shape[0], nprocs+1, dtype=int)
+    chunks = []
+    for i in range(nprocs):
+        chunks.append((component[chunk_inds[i]:chunk_inds[i+1]], pphases[chunk_inds[i]:chunk_inds[i+1]], 
+                     windowduration[chunk_inds[i]:chunk_inds[i+1]]))
+       
+    p = Pool(nprocs)
+    result = p.map(ComputeCornerFrequency, chunks)
+    p.close()
+    p.join()
+    return np.concatenate(result)        
+
+
+def ComputeCornerFrequency(tuples):
+    """Preproccesing of a single waveform before calling fc_Qzero_gridsearch to determine the corner frequency"""
+    
+    fc_arr = np.zeros(tuples[0].shape[0])
+    endfreq=args.maxFreq[0]
+    
+    for i in range(0, fc_arr.size):
+        
+        velocity = tuples[0][i]
+        windowduration = tuples[2][i] * 1.1     # add 10% to not lose information due to the taper
```

Comment:
probably 1.05?
(would be consistent with  ```ind1 = np.max([int(pphase / dt - 0.025 * window_len), 0])```)

Reply:
Done